tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py

```diff
@@ -125,3 +131,54 @@ def fma(a: common.Field, b: common.Field, c: common.Field, /) -> common.Field:
 
     result = fma(field_inp_a, field_inp_b, field_inp_c)
     assert np.allclose(result.ndarray, expected)
+
+
+def test_get_slices_with_named_indices_1d_to_2d_missing_dim_right():
+    field_domain = common.Domain(dims=(IDim,), ranges=(UnitRange(0, 10),))
+    new_domain = common.Domain(dims=(IDim, JDim), ranges=(UnitRange(5, 10), UnitRange(5, 10)))
```

Comment:
let's write these testcases with `Sequence[NamedRange]` and then maybe one case with `Domain` to illustrate `Domain` is a `Sequence[NamedRange]`.